### ISSUE TITLE
Add possibility of splitting root variable based on element type

### ISSUE THREAD

**Issue by v3lik** [multitheftauto/mtasa-blue#749]
Hello. I am suggesting that we could split hidden variable **root** based on element type like:
**root.players, root.vehicles, root.markers, root.peds**, etc. This would work on all element types.

Now, you might say that this can already be achieved in code by looping and etc. but in same way the root variable can be constructed as well. Other good reason for introducing this feature would be plain **simplicity** when writing code and less lines of code needing to be involved.

Also, the code which we would write would be much more **optimized** because sometimes you just don't have the time to loop everything perfectly for a perfect optimization, and if we were to use **e.g** **root.players** on our **event handlers** compared with **root**, that would be an obvious optimization because we'd would involve way less elements in scripting where they're not all needed (since root features every single element type).

This feature could be realized with pure scripting, but it **would not be ideal** as if it were implemented by default. For instance, if we were to implement this in **Lua**, we could not always **guarantee** that every single element of certain type was in our table because we would require all scripts to use our custom type of creating elements in order to keep elements in their tables (overriding functions like createVehicle, createPed, etc.)

So to sum up, improvements are:

1. Simplicity
2. Optimization
3. Less code involved

**Comment by Pirulax** [comment]
Well, root can stay root, and from there you'd have vehicleRoot, pedRoot, etc.. as child elements. 
Calling anything on the root would still work just fine, and vehicleRoot would work just fine.

Would there be any compatibility issues?
